pkg/api/persistentvolume/util.go

```diff
@@ -50,6 +50,10 @@ func VisitPVSecretNames(pv *api.PersistentVolume, visitor func(string) bool) boo
 		if source.ISCSI.SecretRef != nil && !visitor(source.ISCSI.SecretRef.Name) {
 			return false
 		}
+	case source.StorageOS != nil:
+		if source.StorageOS.SecretRef != nil && !visitor(source.StorageOS.SecretRef.Name) && !visitor(source.StorageOS.SecretRef.Namespace) {
```

Comment:
right now, the sole use of VisitPVSecretNames is inferring secret namespace from the PV's `pv.Spec.ClaimRef.Namespace`, because that's how all secret references from a PV were resolved until now.

With this change, the storageos secret namespace is separately specified.

For all the other calls to the visitor, pass `pv.Spec.ClaimRef.Namespace` as the secret namespace (and if pv.Spec.ClaimRef is nil, don't call the visitor)